R/replace_sc_id_with_latest.R

```diff
@@ -49,7 +70,49 @@ replace_sc_id_with_latest <- function(data) {
         .data$social_care_id
       )
     ) %>%
-    dplyr::filter(!is.na(.data$period))
+    dplyr::filter(!is.na(.data$period)) %>%
+    dplyr::select(-tidyselect::ends_with("_old"))
 
   return(return_data)
 }
+
+
+select_linking_id <- function(data) {
+  data %>% dplyr::mutate(linking_id = dplyr::if_else(
+    is.na(.data$anon_chi),
+    paste0("SCID", .data$sending_location, "-", .data$social_care_id),
+    .data$anon_chi
+  ))
+}
+
+add_fy_qtr_from_period <- function(data) {
+  data %>%
+    # create financial_year and financial_quarter variables for sorting
+    dplyr::mutate(
+      financial_year = as.numeric(stringr::str_sub(.data$period, 1, 4)),
+      financial_quarter = stringr::str_sub(.data$period, 6, 6)
+    ) %>%
+    # set financial quarter to 5 when there is only an annual submission -
+    # for ordering periods with annual submission last
+    dplyr::mutate(
+      financial_quarter = dplyr::if_else(
+        is.na(.data$financial_quarter) |
+          .data$financial_quarter == "",
+        "5",
+        .data$financial_quarter
+      )
+    )
+}
+
+which_fy <- function(date, format = c("year", "fyear")) {
+  year <- as.numeric(format(date, "%Y"))
+  month <- as.numeric(format(date, "%m"))
+
+  start_year <- ifelse(month < 4, year - 1, year)
+  end_year <- start_year + 1
+  if (format == "year") {
+    return(start_year)
+  } else {
+    return(paste0(substr(start_year, 3, 4), substr(end_year, 3, 4)))
+  }
+}
```

Comment:
These functions are missing roxygen documentation, could they please be added and updated in the YAML document for the github website please?

I also think they could be taken out of this script into a new script called `extract_fy.R`